core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -429,20 +429,40 @@ private[ui] class JobDataSource(
     val formattedDuration = duration.map(d => UIUtils.formatDuration(d)).getOrElse("Unknown")
     val submissionTime = jobData.submissionTime
     val formattedSubmissionTime = submissionTime.map(UIUtils.formatDate).getOrElse("Unknown")
-    val jobDescription = UIUtils.makeDescription(jobData.description.getOrElse(""),
-      basePath, plainText = false)
+
+    val lastStage = {
+      val stageAttempts = jobData.stageIds.flatMap(store.stageData(_))
```

Comment:
Just a few tiny suggestions for the code, to take or leave. You should be able to omit `(_)`. You can also use `.nonEmpty` below.